{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Importing Jupyter notebook from Insert_py.ipynb\n",
      "Importing Jupyter notebook from Get_py.ipynb\n",
      "Importing Jupyter notebook from Tor.ipynb\n",
      "Importing Jupyter notebook from Proxy_Gatherer.ipynb\n",
      "Importing Jupyter notebook from Init_database.ipynb\n",
      "Importing Jupyter notebook from email_alert.ipynb\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "import datetime\n",
    "import time\n",
    "import requests\n",
    "import nbimporter\n",
    "import random\n",
    "import os\n",
    "import psutil\n",
    "import psycopg2\n",
    "# from multiprocessing import Process, Pool\n",
    "import Insert_py\n",
    "import Get_py\n",
    "import Tor\n",
    "import Proxy_Gatherer as pg\n",
    "import Init_database as indb\n",
    "import email_alert as reminder\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#--------INITIALIZING CONNECTION TO DATABASE--------#\n",
    "db_name=\"db_teld\"\n",
    "user_name=\"postgres\"\n",
    "password = \"ecalscraping\"\n",
    "host=\"127.0.0.1\"\n",
    "port=\"5432\"\n",
    "conn = indb.init_db(db_name, user_name, password, host, port)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#------- Gathering proxies for visiting teld.cn -------##\n",
    "def proxy():\n",
    "    hdr = {'User-Agent':'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36',\n",
    "           'Accept':'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8'}\n",
    "    gatherproxy = pg.ProxyGatherer(hdr)\n",
    "    proxies_xici = gatherproxy.getproxies_xici()\n",
    "    proxies_elite = gatherproxy.getproxies_elite()\n",
    "#     proxies_usproxy = gatherproxy.getproxies_usproxy()\n",
    "    # t = time.time()\n",
    "#     proxiesXici_pool = gatherproxy.proxies_cleanup(proxies_xici) # Could also specify timeout limit you prefer, \n",
    "                                                                    #format--(connect time, read time)\n",
    "#     proxiesElite_pool = gatherproxy.proxies_cleanup(proxies_elite)\n",
    "    proxies = proxies_xici+proxies_elite\n",
    "#     return proxies\n",
    "    return proxies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def write_error(error_log, pillcode, error_msg):\n",
    "    \"\"\"\n",
    "        Input:\n",
    "            error_log - file, write error messages into designated file\n",
    "    \"\"\"\n",
    "    error_log.write(pillcode)\n",
    "    error_log.write(str(error_msg))\n",
    "    error_log.write('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def header_gen():\n",
    "    import numpy as np\n",
    "    app_version = np.random.randint(1,5)\n",
    "    os_version = np.random.randint(0,3)+round(np.random.randint(0,3)*0.1,1)\n",
    "    user_ag = \"Teld/3.{a}.0 (iPhone; iOS 10.{o}; Scale/2.00)\".format(a=app_version,o=os_version)\n",
    "    device = \"app_version=3.{a}.0&os_version=10.{o}&client=ios&device_name=iPhone 6s (A1633/A1688/A1691/A1700)&device_id=38CD294C-CC36-4998-9074-C7F798EE6087&city_code=4403&city_name=%E6%B7%B1%E5%9C%B3&lat=0.000000&lng=0.000000&network=wifi&location_city_name=\".format(a=app_version,o=os_version)\n",
    "    hdr = {\n",
    "        'accept': \"*/*\",\n",
    "        'user-agent': user_ag,\n",
    "        'cookie': \"TELDAppID=\",\n",
    "        'content-length': \"111\",\n",
    "        'connection': \"keep-alive\",\n",
    "        'accept-encoding': \"gzip, deflate\",\n",
    "        'device': device,\n",
    "        'host': \"basesg.teld.cn\",\n",
    "        'content-type': \"application/x-www-form-urlencoded\",\n",
    "        'cache-control': \"no-cache\"\n",
    "    }\n",
    "    return hdr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def ChargeHist_header_gen():\n",
    "    import numpy as np\n",
    "    app_version = np.random.randint(1,5)\n",
    "    os_version = np.random.randint(0,3)+round(np.random.randint(0,3)*0.1,1)\n",
    "    user_ag = \"Teld/3.{a}.0 (iPhone; iOS 10.{o}; Scale/2.00)\".format(a=app_version,o=os_version)\n",
    "    device = \"app_version=3.{a}.0&os_version=10.{o}&client=ios&device_name=iPhone 6s (A1633/A1688/A1691/A1700)&device_id=38CD294C-CC36-4998-9074-C7F798EE6087&city_code=4403&city_name=%E6%B7%B1%E5%9C%B3&lat=0.000000&lng=0.000000&network=wifi&location_city_name=\".format(a=app_version,o=os_version)\n",
    "    hdr = {\n",
    "        'accept': \"*/*\",\n",
    "        'user-agent': user_ag,\n",
    "        'cookie': \"TELDAppID=\",\n",
    "        'content-length': \"250\",\n",
    "        'connection': \"keep-alive\",\n",
    "        'accept-encoding': \"gzip, deflate\",\n",
    "        'device': device,\n",
    "        'host': \"basesg.teld.cn\",\n",
    "        'content-type': \"application/x-www-form-urlencoded\",\n",
    "        'cache-control': \"no-cache\"\n",
    "    }\n",
    "    return hdr\n",
    "ChargeHist_hdr = ChargeHist_header_gen()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5531\n",
      "86\n",
      "8866\n",
      "948\n"
     ]
    }
   ],
   "source": [
    "# ##------- Every once a while runs it update station and pile fixed info -------##\n",
    "# #--------------------------------------------#\n",
    "# #------- Gather all stations  -------# ****NEED TO BE FIXED 09.17.2017\n",
    "# station_info = Get_py.get_allStations(hdr)\n",
    "# print(len(station_info))\n",
    "# for station in station_info:\n",
    "#     stationId = str(station['id'])  \n",
    "# #     print(stationId)\n",
    "#     #------- Gather individual stations details -------#\n",
    "#     station_detailedInfo = Get_py.get_oneStation_APP(hdr, stationId)\n",
    "#     conn = Insert_py.insert_singStation_details(conn, station_detailedInfo)\n",
    "#     #------- Gather individual pile code -------#\n",
    "#     piles_info = Get_py.get_piles(stationId, proxies_pool, hdr)\n",
    "#     conn = Insert_py.insert_pilecode(conn, stationId, piles_info)\n",
    "# #------- Gather inidividual pile details -------#\n",
    "# #------- Request stationid which has user activities from db -------#\n",
    "try:\n",
    "    cur = conn.cursor()\n",
    "    cur.execute(\"select distinct stationid from tbl_app_useractivity;\")\n",
    "    stationWithAct_list = cur.fetchall()\n",
    "    cur.close()\n",
    "except psycopg2.Error as e:\n",
    "    print(e)\n",
    "print(len(stationWithAct_list))\n",
    "# #------- Request pilecode from db for states -------#\n",
    "staid_list = []\n",
    "for staid in stationWithAct_list:\n",
    "    staid_list.append(staid[0])\n",
    "try:\n",
    "    cur = conn.cursor()\n",
    "    sta_list = ['2b3a35d4-cb94-4b8f-a47a-bbb054754066', '1d02640a-7f14-4fc3-baaf-60754a38fff7',\n",
    "                'ac6cb946-5fff-491b-859e-55cc5f701657', 'dd4a3ddc-e1b0-4e2e-b7ca-faef627f4e66',\n",
    "                'b134a159-16bd-4f5d-af1b-cd42599a3bab', '33a3c031-0a3a-417d-ba1b-53549bb08dc2', \n",
    "                'e4758edb-1183-4c51-887b-a52ca7abef3d', '13bdab71-110b-4448-a3c4-5c3113183546']\n",
    "    #1. beijing hangkong jingmi jixie yanjiusuo chongdianzhan --'2b3a35d4-cb94-4b8f-a47a-bbb054754066',\n",
    "    #2. beijing xijiao bingguan chongdianzhan -- '1d02640a-7f14-4fc3-baaf-60754a38fff7\n",
    "    #3. beijing yangguanglicheng jiudian chongdianzhan --'ac6cb946-5fff-491b-859e-55cc5f701657',\n",
    "    #4. beijing rongtao gongsi chongdianzhan -- '32d317a9-3d0c-4458-8357-89a2b3b1cf94'\n",
    "    #5. beijing huairou qu yanshui lu -- 'dd4a3ddc-e1b0-4e2e-b7ca-faef627f4e66'\n",
    "    \n",
    "    #6. shenzhen hengjiang dasha chongdianzhan -- 'b134a159-16bd-4f5d-af1b-cd42599a3bab'\n",
    "    #7. shenzhen yinxing gongyeyuan chongdianzhan -- '33a3c031-0a3a-417d-ba1b-53549bb08dc2'\n",
    "    #8. shenzhe wanke jiulongshan chongdianzhan -- 'e4758edb-1183-4c51-887b-a52ca7abef3d'\n",
    "    #9. shenzhen jingyuan dasha chongdianzhan -- '13bdab71-110b-4448-a3c4-5c3113183546'\n",
    "    stmt = \"select pilecode from tbl_APP_pilecode where staid in %s\"\n",
    "    cur.execute(stmt, (tuple(sta_list),))\n",
    "    \n",
    "#     stmt = \"select pilecode from tbl_APP_pilecode where staid = 'dd4a3ddc-e1b0-4e2e-b7ca-faef627f4e66'\"\n",
    "#     cur.execute(stmt)\n",
    "#     cur.execute(\"select pilecode from tbl_APP_pilecode\")\n",
    "    pillcode_list = cur.fetchall()\n",
    "    cur.close()\n",
    "except psycopg2.Error as e:\n",
    "    print(e)\n",
    "print(len(pillcode_list))\n",
    "\n",
    "# # ------- Requests stationID from db for user charging activities -------#\n",
    "try:\n",
    "    cur = conn.cursor()\n",
    "    cur.execute(\"select stationid from tbl_app_eachstations;\")\n",
    "    station_list = cur.fetchall()\n",
    "    cur.close()\n",
    "except psycopg2.Error as e:\n",
    "    print(e)\n",
    "print(len(station_list))\n",
    "\n",
    "# reportTime = datetime.datetime.now() + datetime.timedelta(minutes=15)\n",
    "# i = 0\n",
    "# for pile in pilecode_list:\n",
    "#     #If encounter error, try one more request with a different proxy\n",
    "#     try:\n",
    "#         pile_detailedInfo, proxies_pool = Get_py.get_onePile_APP(F, hdr, pile[0], proxies_pool, time_out=1000)\n",
    "#     except ValueError as e:\n",
    "#         try:\n",
    "#             pile_detailedInfo, proxies_pool = Get_py.get_onePile_APP(F, hdr, pile[0], proxies_pool, time_out=1000)\n",
    "#         except ValueError as e:\n",
    "#             pile_detailedInfo = None\n",
    "#         except (requests.exceptions.ChunkedEncodingError) as e:\n",
    "#             pile_detailedInfo = None\n",
    "#     except (requests.exceptions.ChunkedEncodingError) as e:\n",
    "#         try:\n",
    "#             pile_detailedInfo, proxies_pool = Get_py.get_onePile_APP(F, hdr, pile[0], proxies_pool, time_out=1000)\n",
    "#         except ValueError as e:\n",
    "#             pile_detailedInfo = None\n",
    "#         except (requests.exceptions.ChunkedEncodingError) as e:\n",
    "#             pile_detailedInfo = None\n",
    "#     if pile_detailedInfo is not None:\n",
    "#         conn = Insert_py.insert_singPile_details(conn, pile_detailedInfo)\n",
    "#     else:\n",
    "#         print(\"Running None Func.\")\n",
    "#         conn = Insert_py.insert_singPile_none(conn, pile_detailedInfo, pile[0])\n",
    "#     i += 1\n",
    "#     if datetime.datetime.now() >= reportTime:\n",
    "#         reportTime = datetime.datetime.now() + datetime.timedelta(minutes=15) \n",
    "#         print(\"Successfully run {} piles\".format(i))\n",
    "# #211 Running None Func.\n",
    "# #71585 piles.\n",
    "# #------- Request Shenzhen charging pilecode from db for states -------#\n",
    "try:\n",
    "    cur = conn.cursor()\n",
    "#     stmt1 = \"select stationid from tbl_app_eachstations where tbl_app_eachstations.location && ST_MakeEnvelope(115.37916, 39.47148, 117.15623, 40.81726, 4326);\"    \n",
    "    stmt1 = \"select stationid from tbl_app_eachstations where tbl_app_eachstations.location && ST_MakeEnvelope(113.85955, 22.50563, 114.17335, 22.69960, 4326) intersect select distinct stationid from tbl_app_useractivity;\"\n",
    "    cur.execute(stmt1)\n",
    "    sta_list = cur.fetchall()\n",
    "    sta_list_clean = []\n",
    "    for sta in sta_list:\n",
    "        sta_list_clean.append(sta[0])\n",
    "    stmt2 = \"select pilecode from tbl_APP_pilecode where staid in %s\"\n",
    "    cur.execute(stmt2, (tuple(sta_list_clean),))\n",
    "    shenzhen_pillcode_list = cur.fetchall()\n",
    "    cur.close()\n",
    "except psycopg2.Error as e:\n",
    "    print(e)\n",
    "print(len(shenzhen_pillcode_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# # #------- Update user activity -------#\n",
    "# i = 0\n",
    "# fail = 0\n",
    "# hour_uni = u'\\u5c0f\\u65f6\\u524d'\n",
    "# minute_uni = u'\\u5206\\u949f\\u524d'\n",
    "# # proxies = proxy()\n",
    "# # proxy_idx = random.randint(0,len(proxies)-1) \n",
    "# # os.environ[\"https_proxy\"]=proxies[proxy_idx]['https']\n",
    "# # os.environ[\"http_proxy\"]=proxies[proxy_idx]['http']\n",
    "# activity_dict = {}\n",
    "# for idx, stationID in enumerate(station_list):\n",
    "#     stationID = stationID[0]\n",
    "# #     if len(proxies) == 1: #proxy pool dried out\n",
    "# #         proxies = proxy()\n",
    "    \n",
    "# #     proxy_idx = random.randint(0,len(proxies)-1)\n",
    "# #     os.environ[\"https_proxy\"]=proxies[proxy_idx]['https']\n",
    "# #     os.environ[\"http_proxy\"]=proxies[proxy_idx]['http']\n",
    "    \n",
    "#     try_num = 2\n",
    "# #     while try_num:\n",
    "# #         if try_num < 2:\n",
    "# #             print(\"Retry\")\n",
    "# #         try:\n",
    "# #             chargeList_detailedInfo = Get_py.get_chargelist_APP(stationID,ChargeHist_hdr)\n",
    "# #         except:\n",
    "# #             print(\"Changing proxy\")\n",
    "# #             proxies.pop(proxy_idx)\n",
    "# #             if len(proxies) == 1:\n",
    "# #                 proxies = proxy()\n",
    "# #                 proxy_idx = random.randint(0,len(proxies)-1)\n",
    "# #             else:\n",
    "# #                 proxy_idx = random.randint(0,len(proxies)-1)\n",
    "# #             os.environ[\"https_proxy\"]=proxies[proxy_idx]['https']\n",
    "# #             os.environ[\"http_proxy\"]=proxies[proxy_idx]['http']\n",
    "# #         try_num -= 1\n",
    "# #         if not try_num:\n",
    "# #             chargeList_detailedInfo=[]\n",
    "# #             fail += 1\n",
    "# #             print('fail +', fail, ' index', idx)\n",
    "                \n",
    "#     try:\n",
    "#         chargeList_detailedInfo = Get_py.get_chargelist_APP(stationID,ChargeHist_hdr)\n",
    "#     except:\n",
    "#         try:\n",
    "# #             proxies.pop(proxy_idx)\n",
    "# #             proxy_idx = random.randint(0,len(proxies)-1)\n",
    "# #             os.environ[\"https_proxy\"]=proxies[proxy_idx]['https']\n",
    "# #             os.environ[\"http_proxy\"]=proxies[proxy_idx]['http']\n",
    "#             chargeList_detailedInfo = Get_py.get_chargelist_APP(stationID,ChargeHist_hdr)\n",
    "#         except:\n",
    "# #             proxies.pop(proxy_idx)\n",
    "#             chargeList_detailedInfo=[]\n",
    "#             fail += 1\n",
    "#             print('fail +', fail, ' index', idx)\n",
    "#             pass\n",
    "#     if len(chargeList_detailedInfo) == 0:\n",
    "#         i += 1\n",
    "#     else:\n",
    "#         activity_dict['{}'.format(stationID)] = chargeList_detailedInfo\n",
    "# print(str(i) + \" stations do not have activity history.\")\n",
    "# print(str(fail) + \" connections failed attempt.\")\n",
    "# print(datetime.datetime.now())\n",
    "# fromaddr = \"296257740.jz@gmail.com\"\n",
    "# psw = \"15710725968crzzq\"\n",
    "# toaddr = \"hustlejanton@berkeley.edu\"\n",
    "# subject=\"Courtesy Email\"\n",
    "# body=\"Program terminated by time - 16046 user activity\"\n",
    "# reminder.email_courtesy(fromaddr, toaddr, psw, subject, body)\n",
    "# print(\"Email sent successfully.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# for stationID in activity_dict.keys():\n",
    "#     chargeList_detailedInfo = activity_dict[stationID]\n",
    "#     for user_act in chargeList_detailedInfo:\n",
    "#         startTime, pileName, carTypeName, userNickName = user_act_AIclean(user_act)\n",
    "#         conn = Insert_py.insert_userActivity_info(conn, startTime, pileName, stationID, carTypeName, userNickName)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def pillState_scrap(pillcode): \n",
    "    \"\"\"\n",
    "        Input:\n",
    "            pillCode - string, each charging pile has its own pile code\n",
    "        Output:\n",
    "            pill_detailedInfo - dictionary of pill state details\n",
    "    \"\"\"\n",
    "    hdr = header_gen()\n",
    "    time_out = 20\n",
    "    F = None\n",
    "    proxies_pool = None\n",
    "    #If encounter error, try one more request with a different proxy\n",
    "    try:\n",
    "        pill_detailedInfo, proxies_pool = Get_py.get_onePill_APP(hdr, pillcode, F, proxies_pool, time_out=time_out)\n",
    "#     except ValueError as e:\n",
    "#         if F is not None:\n",
    "#             write_error(F, pillcode, e)\n",
    "#         try:\n",
    "#             pill_detailedInfo, proxies_pool = Get_py.get_onePill_APP(hdr, pillcode, F, proxies_pool, time_out=time_out)\n",
    "#         except ValueError as e:\n",
    "#             if F is not None:\n",
    "#                 write_error(F, pillcode, e)\n",
    "#             pill_detailedInfo = None\n",
    "#         except (requests.exceptions.ChunkedEncodingError) as e:\n",
    "#             if F is not None:\n",
    "#                 write_error(F, pillcode, e)\n",
    "# #             else:\n",
    "#             pill_detailedInfo = None\n",
    "#     except (requests.exceptions.ChunkedEncodingError) as e:\n",
    "#         if F is not None:\n",
    "#             write_error(F, pillcode, e)\n",
    "#         try:\n",
    "#             pill_detailedInfo, proxies_pool = Get_py.get_onePill_APP(hdr, pillcode, F, proxies_pool, time_out=time_out)\n",
    "#         except ValueError as e:\n",
    "#             if F is not None:\n",
    "#                 write_error(F, pillcode, e)\n",
    "#             pill_detailedInfo = None\n",
    "#         except (requests.exceptions.ChunkedEncodingError) as e:\n",
    "#             if F is not None:\n",
    "#                 write_error(F, pillcode, e)\n",
    "# #         else:\n",
    "#             pill_detailedInfo = None\n",
    "# #     return pill_detailedInfo, proxies_pool\n",
    "    except:\n",
    "        pill_detailedInfo = None\n",
    "    return pill_detailedInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def insert_state(outputs):\n",
    "    for pill_detailedInfo in outputs:\n",
    "        if pill_detailedInfo is not None:\n",
    "            db_name=\"db_teld\"\n",
    "            user_name=\"postgres\"\n",
    "            password = \"ecalscraping\"\n",
    "            host=\"127.0.0.1\"\n",
    "            port=\"5432\"\n",
    "            conn = indb.init_db(db_name, user_name, password, host, port)\n",
    "            pillCode = str(pill_detailedInfo['pillCode'])\n",
    "            stateCode = str(pill_detailedInfo['stateCode'])\n",
    "            stateCodeV3 = str(pill_detailedInfo['stateCodeV3'])\n",
    "            if pill_detailedInfo['stateName'] is not None:\n",
    "                stateName = str(pill_detailedInfo['stateName'].encode('utf-8'))\n",
    "            else:\n",
    "                stateName = ''\n",
    "            if pill_detailedInfo['stateNameV3'] is not None:\n",
    "                stateNameV3 = str(pill_detailedInfo['stateNameV3'].encode('utf-8'))\n",
    "            else:\n",
    "                stateNameV3 = ''\n",
    "            conn = Insert_py.insert_state_info(conn, pillCode, stateCode, stateName, stateCodeV3, stateNameV3)\n",
    "            conn.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## ----- Test for single looping of multiprocessing post_request and insert -----##\n",
    "\n",
    "# pid = os.getpid()\n",
    "# py = psutil.Process(pid)\n",
    "# memoryUse = py.memory_info()[0]/2.**20 #in MB\n",
    "# print('Before runnning script, memory use:', memoryUse, 'MB')\n",
    "# # loop_num = 1\n",
    "# # max_proc =10000\n",
    "# pool = Pool(200)\n",
    "# t = time.time()\n",
    "# if __name__ == '__main__':\n",
    "#     results = [pool.apply_async(pillState_scrap, (pillcode[0],)) for pillcode in pillcode_list]\n",
    "#     print(len(results))\n",
    "#     outputs = [p.get() for p in results]\n",
    "# print(\"success, time costs is \" + str(time.time()-t)) \n",
    "\n",
    "# pid = os.getpid()\n",
    "# py = psutil.Process(pid)\n",
    "# memoryUse = py.memory_info()[0]/2.**20 #in MB\n",
    "# print('After runnning script,memory use:', memoryUse, 'MB')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4403060016101\n",
      "4403060072102\n",
      "4403060072110\n",
      "4403060072112\n",
      "4403060072104\n",
      "4403060072106\n",
      "4403060072107\n",
      "4403060072113\n",
      "4403060072111\n",
      "4403060072103\n",
      "4403060072108\n",
      "4403060072109\n",
      "4403060072101\n",
      "4403060072115\n",
      "4403060072105\n",
      "4403060072114\n"
     ]
    }
   ],
   "source": [
    "for pillcode in pillcode_list[70:]:\n",
    "    print(pillcode[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# os.environ[\"https_proxy\"]=\"https://165.84.167.54:8080\"\n",
    "# os.environ[\"http_proxy\"]=\"http://165.84.167.54:8080\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1f6810>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e2098d0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1a9650>, 'Connection to 103.74.246.127 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1a9990>, 'Connection to 202.77.127.98 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Takes 380.876028061 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 504 Host not found',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e209a90>, 'Connection to 101.37.79.125 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e209a90>, 'Connection to 183.51.191.251 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 407 Proxy Authentication Required',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283ea123d0>, 'Connection to 185.85.21.6 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c07d0>, 'Connection to 122.183.139.101 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 403 Forbidden',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0610>, 'Connection to 114.215.95.188 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0e90>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0710>, 'Connection to 183.159.86.234 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e446490>, 'Connection to 153.149.169.243 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e279a90>, 'Connection to 185.17.132.66 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 513.711566925 seconds\n",
      "Missing 11 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0d10>, 'Connection to 147.75.113.108 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 403 Forbidden',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e275e90>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e275e90>, 'Connection to 139.224.80.139 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e272f50>, 'Connection to 5.11.70.196 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e275e90>, 'Connection to 27.46.22.198 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 504 Gateway Time-out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 399.528258085 seconds\n",
      "Missing 8 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e209510>, 'Connection to 104.46.34.250 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429150>, 'Connection to 223.241.78.220 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0350>, 'Connection to 183.159.86.234 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0ad0>, 'Connection to 171.13.3.6 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e279fd0>, 'Connection to 195.49.200.154 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Timeout occured\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 503 Too many open connections',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e431d50>: Failed to establish a new connection: [Errno -2] Name or service not known',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e431350>, 'Connection to 188.120.245.252 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e431550>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e4488d0>, 'Connection to 153.149.169.239 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 559.656609058 seconds\n",
      "Missing 12 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c00d0>, 'Connection to 168.195.210.163 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0e50>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0e10>, 'Connection to 109.87.127.175 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 503 Too many open connections',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e2753d0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1a2910>, 'Connection to 153.149.169.240 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 504 Gateway Time-out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1a2190>, 'Connection to 203.84.142.95 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 410.217774153 seconds\n",
      "Missing 8 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0390>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 500 Internal Privoxy Error',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e272410>, 'Connection to 13.57.207.226 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e272090>, 'Connection to 42.7.26.21 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Invalid response from tunnel request',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e272b50>, 'Connection to 183.159.93.120 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Takes 336.88627696 seconds\n",
      "Missing 8 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264e90>, 'Connection to 153.149.169.203 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264dd0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264150>, 'Connection to 149.202.34.104 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264810>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429090>, 'Connection to 93.188.162.77 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429410>, 'Connection to 171.13.3.6 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e279fd0>, 'Connection to 159.89.198.67 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Takes 475.313950062 seconds\n",
      "Missing 8 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429f90>, 'Connection to 128.199.199.41 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e272b10>, 'Connection to 206.189.87.18 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e272c10>, 'Connection to 110.77.232.83 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 396.585886002 seconds\n",
      "Missing 5 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1f6210>, 'Connection to 114.215.83.184 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0350>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e279790>, 'Connection to 218.72.67.49 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e2798d0>, 'Connection to 109.195.60.33 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e272890>, 'Connection to 159.65.110.167 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Takes 397.185134888 seconds\n",
      "Missing 5 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e275490>, 'Connection to 177.89.149.121 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3257d0>, 'Connection to 110.77.203.59 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1f6090>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Takes 303.225520134 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264410>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 504 Gateway Time-out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429710>, 'Connection to 194.187.216.228 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e325990>, 'Connection to 109.87.127.175 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33ce90>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33c9d0>, 'Connection to 181.196.145.106 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e325910>, 'Connection to 92.53.73.138 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 462.153027058 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264f10>, 'Connection to 153.149.169.240 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33cd90>, 'Connection to 13.92.196.150 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e35af50>, 'Connection to 60.177.225.98 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264450>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e9ea910>, 'Connection to 202.77.127.98 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 402.153327942 seconds\n",
      "Missing 6 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c07d0>, 'Connection to 202.191.121.171 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c03d0>, 'Connection to 139.224.80.139 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264bd0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e2644d0>, 'Connection to 183.184.193.200 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264990>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264190>, 'Connection to 153.149.171.166 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264bd0>, 'Connection to 122.183.139.98 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0f50>, 'Connection to 35.194.85.89 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c00d0>, 'Connection to 40.115.241.124 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Takes 452.457114935 seconds\n",
      "Missing 9 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3330d0>, 'Connection to 114.215.95.188 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 403 Forbidden',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e333ed0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e333f50>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3332d0>, 'Connection to 94.75.174.75 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33cfd0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e2642d0>, 'Connection to 122.183.137.190 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0150>, 'Connection to 171.37.159.138 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Invalid response from tunnel request',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing proxy\n",
      "Takes 331.681015968 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5bb350>, 'Connection to 188.120.245.252 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3332d0>: Failed to establish a new connection: [Errno -2] Name or service not known',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Timeout occured\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c0a90>, 'Connection to 185.17.132.66 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c0390>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33c610>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264f50>, 'Connection to 46.250.22.85 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 407 Proxy Authentication Required',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c0150>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e1c0c90>, 'Connection to 153.149.169.200 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 405.266232014 seconds\n",
      "Missing 8 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e333950>, 'Connection to 114.113.126.83 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e337fd0>, 'Connection to 218.72.66.162 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264c50>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Takes 296.310839176 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c0610>, 'Connection to 206.189.87.18 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e325c90>, 'Connection to 171.13.3.6 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e325490>, 'Connection to 147.75.113.108 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 403 Forbidden',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3378d0>, 'Connection to 110.73.32.164 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Takes 458.002781868 seconds\n",
      "Missing 8 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e431950>, 'Connection to 153.149.169.206 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e485410>, 'Connection to 13.57.207.226 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e35a950>, 'Connection to 46.98.188.108 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5bb310>, 'Connection to 101.37.79.125 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5bb190>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e333bd0>, 'Connection to 171.37.159.138 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e333cd0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3dd390>, 'Connection to 169.239.126.150 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Takes 635.032468081 seconds\n",
      "Missing 9 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5bb490>, 'Connection to 42.7.26.21 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3dded0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e4cfb10>, 'Connection to 114.25.52.191 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 300.978155136 seconds\n",
      "Missing 6 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3332d0>, 'Connection to 114.113.126.83 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 403 Forbidden',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3dd450>, 'Connection to 42.104.84.109 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33ced0>, 'Connection to 218.72.67.220 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c09d0>, 'Connection to 27.46.74.25 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c0750>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c0550>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 435.168073893 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e327f90>, 'Connection to 42.7.26.21 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33c090>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33c450>, 'Connection to 94.75.174.75 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e387f50>: Failed to establish a new connection: [Errno -2] Name or service not known',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e387950>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33cc10>, 'Connection to 40.115.241.124 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "('Connection aborted.', BadStatusLine(\"''\",))\n",
      "failed\n",
      "Changing proxy\n",
      "Takes 354.650023937 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264710>, 'Connection to 46.42.38.171 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e333d10>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e4299d0>, 'Connection to 13.92.196.150 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429150>, 'Connection to 114.215.95.188 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 503 Forwarding failure',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Invalid response from tunnel request',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429a90>, 'Connection to 52.207.93.207 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 405.524923086 seconds\n",
      "Missing 7 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e5c0fd0>, 'Connection to 114.215.95.188 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e431910>, 'Connection to 153.149.169.203 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 411.004554033 seconds\n",
      "Missing 5 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e35a290>, 'Connection to 153.149.169.215 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e35a490>, 'Connection to 60.177.227.34 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Timeout occured\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e35af50>, 'Connection to 183.184.193.200 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e3ed290>, 'Connection to 38.96.9.236 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 504 Gateway Time-out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e264b10>, 'Connection to 101.37.79.125 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e325b90>, 'Connection to 110.77.203.59 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 403 Forbidden',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 504 Gateway Time-out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33cb10>, 'Connection to 153.149.171.166 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33c810>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33ca90>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 407 Proxy Authentication Required',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33ce50>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e33c990>, 'Connection to 114.113.126.83 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e4851d0>, 'Connection to 153.149.169.198 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 500 Internal Privoxy Error',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e485390>, 'Connection to 218.72.67.220 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 630.175215006 seconds\n",
      "Missing 13 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e4291d0>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('select timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Takes 295.088676929 seconds\n",
      "Missing 5 of pills data\n",
      "sleeping...\n",
      "waking up!\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e9ea610>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e9ea4d0>, 'Connection to 139.224.80.139 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e9eac50>, 'Connection to 183.51.191.251 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', error('Tunnel connection failed: 407 Proxy Authentication Required',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e4295d0>, 'Connection to 114.215.83.184 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429a10>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429510>, 'Connection to 41.76.150.154 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ConnectTimeoutError(<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429690>, 'Connection to 27.46.74.25 timed out. (connect timeout=20)'))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f283e429750>: Failed to establish a new connection: [Errno 111] Connection refused',)))\n",
      "failed\n",
      "Changing proxy\n",
      "HTTPSConnectionPool(host='basesg.teld.cn', port=443): Max retries exceeded with url: /api/invoke?SID=BaseApi-App0304_GetTerminalDetails (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))\n",
      "failed\n",
      "Changing proxy\n",
      "Retry\n",
      "Changing proxy\n",
      "Takes 423.250110865 seconds\n",
      "Missing 10 of pills data\n",
      "sleeping...\n"
     ]
    }
   ],
   "source": [
    "# pool = Pool(processes=2)\n",
    "# startTime = datetime.datetime.now()\n",
    "endTime = datetime.datetime.now() + datetime.timedelta(hours=96)\n",
    "# proxyUpdateTime = datetime.datetime.now() + datetime.timedelta(hours=3)\n",
    "# tmpproxy = {'http': 'http://101.78.219.178:8080', 'https': 'https://101.78.219.178:8080'}\n",
    "# os.environ[\"https_proxy\"]=tmpproxy['https']\n",
    "# os.environ[\"http_proxy\"]=tmpproxy['http']\n",
    "proxies = proxy()\n",
    "proxy_idx = random.randint(0,len(proxies)-1) \n",
    "os.environ[\"https_proxy\"]=proxies[proxy_idx]['https']\n",
    "os.environ[\"http_proxy\"]=proxies[proxy_idx]['http']\n",
    "\n",
    "fromaddr = \"296257740.jz@gmail.com\"\n",
    "psw = \"15710725968crzzq\"\n",
    "toaddr = \"hustlejanton@berkeley.edu\"\n",
    "subject=\"Courtesy Email\"\n",
    "try:\n",
    "    while True:\n",
    "        if datetime.datetime.now() >= endTime:\n",
    "            body=\"Program terminated by time - 16043\"\n",
    "            reminder.email_courtesy(fromaddr, toaddr, psw, subject, body)\n",
    "            break\n",
    "    #     Tor.renewTor()\n",
    "    #     Tor.connectTor()\n",
    "    #     Tor.showmyIp()\n",
    "        print(\"waking up!\")\n",
    "    #     if datetime.datetime.now() >= proxyUpdateTime: #hit update time\n",
    "    #         proxies = proxy()\n",
    "    #         proxyUpdateTime = datetime.datetime.now() + datetime.timedelta(hours=3)\n",
    "        if len(proxies) <= 1: #proxy pool dried out before next update time\n",
    "            proxies = proxy()\n",
    "            proxy_idx = random.randint(0,len(proxies)-1) \n",
    "            os.environ[\"https_proxy\"]=proxies[proxy_idx]['https']\n",
    "            os.environ[\"http_proxy\"]=proxies[proxy_idx]['http']\n",
    "        t = time.time()\n",
    "        outputs=[]\n",
    "    #     results = [pool.apply_async(pillState_scrap, (pillcode[0],)) for pillcode in pillcode_list]\n",
    "    #     outputs = [p.get() for p in results]\n",
    "        missing_data = 0\n",
    "        for pillcode in pillcode_list:\n",
    "    #         Tor.connectTor()\n",
    "            try_num = 2\n",
    "            while try_num:\n",
    "                if try_num < 2:\n",
    "                    print(\"Retry\")\n",
    "                pill_detailedInfo = pillState_scrap(pillcode[0])\n",
    "                if pill_detailedInfo is not None:\n",
    "                    outputs.append(pill_detailedInfo)\n",
    "                    break\n",
    "                if pill_detailedInfo is None:\n",
    "                    print(\"Changing proxy\")\n",
    "                    proxies.pop(proxy_idx)\n",
    "                    if len(proxies) == 1:\n",
    "                        proxies = proxy()\n",
    "                        proxy_idx = random.randint(0,len(proxies)-1)\n",
    "                    else:\n",
    "                        proxy_idx = random.randint(0,len(proxies)-1)\n",
    "                    os.environ[\"https_proxy\"]=proxies[proxy_idx]['https']\n",
    "                    os.environ[\"http_proxy\"]=proxies[proxy_idx]['http']\n",
    "                try_num -= 1\n",
    "                if not try_num:\n",
    "                    missing_data += 1\n",
    "        if endTime - datetime.datetime.now() <= datetime.timedelta(hours=1):\n",
    "            body=\"Program less than one hour left. -16043\"\n",
    "            reminder.email_courtesy(fromaddr, toaddr, psw, subject, body)\n",
    "        print(\"Takes \"+str(time.time()-t)+\" seconds\")\n",
    "        print(\"Missing {} of pills data\".format(missing_data))\n",
    "        insert_state(outputs)\n",
    "        print(\"sleeping...\")\n",
    "        time.sleep(random.randint(100, 400))\n",
    "except Exception as e: \n",
    "    print(e)\n",
    "    print(datetime.datetime.now())\n",
    "    body=\"Program terminated by error - 16043\"\n",
    "    reminder.email_courtesy(fromaddr, toaddr, psw, subject, body)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# ## pid = os.getpid()\n",
    "# py = psutil.Process(pid)\n",
    "# memoryUse = py.memory_info()[0]/2.**20 #in MB\n",
    "# print('Before runnning script, memory use:', memoryUse, 'MB')\n",
    "# pool = Pool(100)\n",
    "# t = time.time()\n",
    "# if __name__ == '__main__':\n",
    "#     results = [pool.apply_async(insert_state, (pill_detailedInfo,)) for pill_detailedInfo in outputs]\n",
    "#     print(len(results))\n",
    "#     outputs = [p.get() for p in results]\n",
    "# print(\"success, time costs is \" + str(time.time()-t)) \n",
    "# pid = os.getpid()\n",
    "# py = psutil.Process(pid)\n",
    "# memoryUse = py.memory_info()[0]/2.**20 #in MB\n",
    "# print('After runnning script, memory use:', memoryUse, 'MB')## pid = os.getpid()\n",
    "# py = psutil.Process(pid)\n",
    "# memoryUse = py.memory_info()[0]/2.**20 #in MB\n",
    "# print('Before runnning script, memory use:', memoryUse, 'MB')\n",
    "# pool = Pool(100)\n",
    "# t = time.time()\n",
    "# if __name__ == '__main__':\n",
    "#     results = [pool.apply_async(insert_state, (pill_detailedInfo,)) for pill_detailedInfo in outputs]\n",
    "#     print(len(results))\n",
    "#     outputs = [p.get() for p in results]\n",
    "# print(\"success, time costs is \" + str(time.time()-t)) \n",
    "# pid = os.getpid()\n",
    "# py = psutil.Process(pid)\n",
    "# memoryUse = py.memory_info()[0]/2.**20 #in MB\n",
    "# print('After runnning script, memory use:', memoryUse, 'MB')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
